File: week02/src/boj_10773_wed/boj_10773_samyoahri.java
package boj_10773_wed;

import java.io.*;
import java.util.Stack;

public class boj_10773_samyoahri {
    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter out = new BufferedWriter(new OutputStreamWriter(System.out));

        Stack<Integer> stack = new Stack<>();

        int k = Integer.parseInt(in.readLine());

        for(int i = 0; i < k; i++){
            int num = Integer.parseInt(in.readLine());
            if(num==0){
                if(stack.isEmpty())
                    out.write("Error");
                stack.pop();
            }
            else{
                stack.push(num);
            }
        }
        int sum = 0;
        while(!stack.isEmpty()){
            int stackNum = stack.pop();
            sum+=stackNum;
        }
//        StringBuilder sb = new StringBuilder();
//        String res = sb.append(sum);
        String res = Integer.toString(sum);
        out.write(res);
//        System.out.println(res);
        out.flush();
        out.close();
    }
}

File: 2023/Nov_2023/week48/boj_10773_samyoahri.cpp
#include<iostream>
#include<stack>

using namespace std;

int main(){
    int N = 0;
    cin >> N;
    int num = -1;
    stack<int> stack;
    for(int i = 0; i < N; ++i){
        cin >> num;
        if(num==0 && !stack.empty())    stack.pop();
        else                            stack.push(num);
    }
    int sum = 0;
    while(!stack.empty()){
        sum += stack.top();
        stack.pop();
    }
    cout << sum << "\n";
    return 0;
}